aws/rust-runtime/aws-config/src/profile/credentials/repr.rs

```diff
@@ -141,22 +145,29 @@ pub fn resolve_chain<'a>(
                 break BaseProvider::AccessKey(static_credentials);
             }
         }
-        let next_profile = match chain_provider(profile) {
-            // this provider wasn't a chain provider, reload it as a base provider
-            None => {
+
+        let next_profile = {
+            if let Some(role_provider) = role_arn_from_profile(profile) {
```

Comment:
this whole change is great! can you add one comment here explaining that the existence of a `role_arn` is the only signal that multiple profiles will be chained?